ngraph_creator/operations/src/FullyConnected.cpp

```diff
@@ -0,0 +1,95 @@
+#include <FullyConnected.hpp>
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+FullyConnected::FullyConnected(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+// Supports only FP32 input. Will add support for QUANT8 through decompose node
+// once the vpu and gna plugin support if confirmed
+bool FullyConnected::validate() {
+    auto input0 = getInputOperand(0);
+    auto input2 = getInputOperand(2);
+    auto output0 = getOutputOperand(0);
+
+    if (input0.type != OperandType::TENSOR_FLOAT32) {
+        ALOGD("%s: Input operand 0 is not of type FP32. Unsupported operation", __func__);
+        return false;
+    }
+
+    if (output0.type != OperandType::TENSOR_FLOAT32) {
+        ALOGD("%s Output operand 0 is not of type FP32 / I32. Unsupported operation", __func__);
+        return false;
+    }
+
+    // Input 2 should be of same type of Input 0
+    if (input0.type != input2.type) {
+        return false;
+    }
+
+    if (input0.dimensions[0] == 0) {
+        ALOGE("%s Batch size of 0 is not supported", __func__);
+        return false;
+    }
+
+    if (input0.dimensions.size() < 2) {
+        ALOGE("%s Invalid input parameter dimensions!!!", __func__);
+        return false;
+    }
+
+    ALOGD("%s succeeded", __func__);
+    return true;
+}
+
+std::shared_ptr<ngraph::Node> FullyConnected::createNode() {
+    ALOGD("%s", __func__);
+
+    std::shared_ptr<ngraph::Node> inputNode = getInputNode(0, false);
+    std::shared_ptr<ngraph::Node> weightsNode = getInputNode(1, false);
+    std::shared_ptr<ngraph::Node> biasNode, multiplyNode, addNode, activationNode;
+
+    auto inputDims = getInputOperand(0).dimensions;
+    auto weightDims = getInputOperand(1).dimensions;
+    auto biasDims = getInputOperand(2).dimensions;
+
+    if ((inputDims[1] != weightDims[1])) {
+        ALOGE("Total input dimenions: %d", getNumberOfElements(inputDims));
```

Comment:
Can you change these to ALOGD's or ALOGI's ?

Reply:
Removed as these are not necessary.